Various Improvements

p1u3o opened the pull request:
I've been using my own PHP manifest for quite a few months and would rather use this one, so I've added some changes onto this extension that should make it a lot better to use with VSCode and other Flatpak IDEs. 

Edits is on so please adjust as you see fit.

**Added**
- README - so people know how to use it.
- ~~LICENSE.txt - that gets copied into the SDK (similar to .NET SDK).~~
- Licenses get included in the runtime, IANAL but distributions do this and the licenses imply you need to.
- enable.sh - so the VSCode Flatpak can eventually automatically add the SDK to path (similar to .NET SDK)
- XDebug - adds php debugging and profiling
- Custom INI support - php will scan .ini files in ~~/.config/php/7.3/ini~~ /var/config/php/7.3/ini
- --enable-mysqli

**Updated**
- PHP 7.3.5 => 7.3.7

**Fixed**
- Tidied compile options
- --with-curl was missing
- --with-gd was missing
- --with-libzip broken

gasinvein commented on org.freedesktop.Sdk.Extension.php73.json:
```diff
@@ -98,28 +96,54 @@
-				"--with-zlib",
-				"--with-openssl"
+				"--with-argon2",
+				"--with-config-file-scan-dir=/usr/lib/sdk/php73/lib/ini:${HOME}/.config/php/7.3/ini",
```
Flatpak keeps config in `~/.var/appid/config`, `~/.config` is neither exposed nor persistent by default.
```suggestion
				"--with-config-file-scan-dir=/usr/lib/sdk/php73/lib/ini:${XDG_CONFIG_HOME}/php/7.3/ini",
```

p1u3o replied:
I was a bit worried about doing that because that wouldn't allow the php configuration to be shared between different IDEs.

gasinvein replied:
`~/.config` wouldn't be shared either, unless each IDE has access to host's `~/.config`. Also with per-IDE config, it could still be shared between IDEs by explicitly adding permission for it, via e.g. `--filesystem=xdg-config/php`.

p1u3o replied:
I've updated it to use /var/config, good to you?

gasinvein replied:
Probably we should also add a per-app, systemd-wide path here, like `/app/etc/php`, so bundled (copied from sdk extension to app) PHP would work, too.

p1u3o replied:
Added that path.

gasinvein replied:
You can achieve cross-flatpak config by manually adding override `--filesystem=xdg-config/php` for you IDEs.

p1u3o replied:
Yeh. Is this all good now?

gasinvein replied:
I believe so, but we can't say for sure until all covered cases are tested.

p1u3o replied:
I'll mark this as resolve, if someone wants to set a shared custom location, they can just add PHP_INI_SCAN_DIR as an environment variable to the flatpak.

gasinvein commented on org.freedesktop.Sdk.Extension.php73.json:
```diff
@@ -98,28 +96,54 @@
+		},
+		{
+			"name":"xdebug",
+			"buildsystem":"simple",
```
What's the reason to use `simple` buildsystem here, instead of the default, `autotools`?

p1u3o replied:
The need to run phpize before ./configure is run.

I've switched it to using autotools with a workaround, does that work better?

gasinvein replied:
You can run something before the build stage by adding a source node of type `shell`, e.g.
```json
{
	"type": "shell",
	"commands": [
		"${FLATPAK_DEST}/bin/phpize"
	]
}
```

p1u3o replied:
Changed.

gasinvein commented on org.freedesktop.Sdk.Extension.php73.json:
```diff
@@ -98,28 +96,60 @@
-				"--with-zlib",
-				"--with-openssl"
+				"--with-argon2",
+				"--with-config-file-scan-dir=${FLATPAK_DEST}/lib/etc:/app/etc/php:/var/config/php/7.3/ini",
```
Why `${FLATPAK_DEST}/lib/etc` and not just `${FLATPAK_DEST}/etc`?

p1u3o replied:
Because the default php.ini is in lib.

I thought about putting it in etc/php to match with /app/etc/php, but then the path would be /usr/lib/sdk/php73/etc/php

Just trying to find some uniformity.

gasinvein replied:
What's wrong with `/usr/lib/sdk/php73/etc/php`? It's immutable anyway.

p1u3o replied:
Thinking about it again, it is better.

p1u3o commented:
> I don't think PHP license is needed in this repo. It's already referenced in the appdata.

I was interpreting this line here.

https://github.com/php/php-src/blob/master/LICENSE#L13 seemed like a valid reason to put the license in with the sdk. 

The MIT has something similar
https://github.com/composer/composer/blob/master/LICENSE#L10

gasinvein commented:
IANAL, but it seems like to fulfill this condition you have to include 
the licence into the flatpak bundle, e.g. put it somewhere in 
/usr/lib/sdk/php73/. This repo doesn't actually contain PHP.

p1u3o commented:
It's gets copied into /usr/lib/sdk/php73

I agree it's not needed in the repository, I could adjust the manifest to copy the licenses from their respective sources, copying Fedora's FHS, placing them in /usr/lib/sdk/php73/share/licenses/<package>, e.g. the Composer RPM installs /usr/share/licenses/composer/LICENSE (which itself is a symlink to /usr/share/composer/license).

p1u3o commented:
Updated this branch.

Fixed libzip not being used, thus --with-libzip not doing anything
Licenses are now fetched and included at build time from their respective sources.
